### PR TITLE
fix(bench): bring in benchmark fixes from main

### PR DESCRIPTION
**Pull request by joshua-goldstein**
Cherry pick of #1699 and #1741.

### CODE REVIEW

File: badger/cmd/pick_table_bench.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime/pprof"
+	"sort"
+	"testing"
+
+	"github.com/dgraph-io/badger/v3"
```
Comment (mangalaman93) [inline badger/cmd/pick_table_bench.go:27]: fix import order

File: badger/cmd/pick_table_bench.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+func pickTableBench(cmd *cobra.Command, args []string) error {
+	opt := badger.DefaultOptions(sstDir).
+		WithValueDir(vlogDir).
+		WithReadOnly(pickOpts.readOnly)
+	fmt.Printf("Opening badger with options = %+v\n", opt)
+	db, err := badger.OpenManaged(opt)
+	if err != nil {
+		return y.Wrapf(err, "unable to open DB")
+	}
+	defer db.Close()
```
Comment (mangalaman93) [inline badger/cmd/pick_table_bench.go:70]: let's log the warning here in case close fails

File: badger/cmd/pick_table_bench.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+// Sorts the boundaries and creates mock table out of them.
+func genTables(boundaries [][]byte) []*table.Table {
+	buildTable := func(k1, k2 []byte) *table.Table {
+		opts := table.Options{
+			ChkMode: options.NoVerification,
+		}
+		b := table.NewTableBuilder(opts)
+		defer b.Close()
```
Comment (mangalaman93) [inline badger/cmd/pick_table_bench.go:193]: same here

**Review by mangalaman93 — CHANGES_REQUESTED** [review]
mostly looks good, a few nits.

File: badger/cmd/pick_table_bench.go (hunk 3 above)
Comment (joshua-goldstein) [reply badger/cmd/pick_table_bench.go:194]: Closing a table builder doesn't return an error

File: badger/cmd/pick_table_bench.go (hunk 1 above)
Comment (joshua-goldstein) [reply badger/cmd/pick_table_bench.go:27]: how so?  `go fmt` didn't make any changes

File: badger/cmd/pick_table_bench.go (hunk 2 above)
Comment (joshua-goldstein) [reply badger/cmd/pick_table_bench.go]: replaced with
```	
defer func() {
y.Check(db.Close())
}()
```

File: badger/cmd/pick_table_bench.go (hunk 1 above)
Comment (mangalaman93) [reply badger/cmd/pick_table_bench.go:27]: It is just a good convention to separate dgraph and non-dgraph imports. First non-dgraph imports and then dgraph imports. Look at any other file.

Comment (mangalaman93) [reply badger/cmd/pick_table_bench.go:27]: @joshua-goldstein the order needs to be opposite. First non-dgraph imports and then dgraph imports

Comment (joshua-goldstein) [reply badger/cmd/pick_table_bench.go:29]: done

**Review by mangalaman93 — APPROVED** [review]

**Pull request merged**